Fails in Python 3.13 with TypeError: typing.Annotated cannot be used with issubclass()

**Describe the bug**

Python 3.13 modified the `typing.Annotated` class so that it no longer supports `isinstance` or `issubclass`. This breaks `dataclasses_avroschema.utils.is_annotated`, which in turn causes a lot of other failures. Example backtrace:

```
src/safir/metrics/_models.py:82: in validate_structure
    schema = tmp_cls.avro_schema_to_python()
../.nox/test-usr-bin-python3-13/lib/python3.13/site-packages/dataclasses_avroschema/main.py:108: in avro_schema_to_python
    avro_schema = cls.generate_schema()
../.nox/test-usr-bin-python3-13/lib/python3.13/site-packages/dataclasses_avroschema/main.py:61: in generate_schema
    cls._parser = cls._generate_parser()
../.nox/test-usr-bin-python3-13/lib/python3.13/site-packages/dataclasses_avroschema/pydantic/main.py:96: in _generate_parser
    return PydanticParser(type=cls, parent=cls._parent or cls)
../.nox/test-usr-bin-python3-13/lib/python3.13/site-packages/dataclasses_avroschema/pydantic/parser.py:22: in __init__
    super().__init__(type, parent)
../.nox/test-usr-bin-python3-13/lib/python3.13/site-packages/dataclasses_avroschema/parser.py:36: in __init__
    self.fields = self.parse_fields(exclude=exclude)
../.nox/test-usr-bin-python3-13/lib/python3.13/site-packages/dataclasses_avroschema/pydantic/parser.py:50: in parse_fields
    AvroField(
../.nox/test-usr-bin-python3-13/lib/python3.13/site-packages/dataclasses_avroschema/fields/fields.py:925: in field_factory
    if utils.is_annotated(native_type):
../.nox/test-usr-bin-python3-13/lib/python3.13/site-packages/dataclasses_avroschema/utils.py:86: in is_annotated
    return origin is not None and isinstance(origin, type) and issubclass(origin, Annotated)  # type: ignore[arg-type]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = typing.Annotated, cls = <class 'types.UnionType'>

    def __subclasscheck__(self, cls):
>       raise TypeError(f"{self} cannot be used with issubclass()")
E       TypeError: typing.Annotated cannot be used with issubclass()
```
**To Reproduce**

Run the dataclasses-avroschema test suite under Python 3.13.

**Expected behavior**

Test suite passes under Python 3.13.

PR coming in just a moment, based on code from Pydantic.

--- a/.github/workflows/tests.yaml
+++ b/.github/workflows/tests.yaml
@@ -17,7 +17,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.9", "3.10", "3.11", "3.12"]
+        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13"]
     steps:
       - uses: actions/checkout@v4
       - name: Setup pythonß

--- a/dataclasses_avroschema/utils.py
+++ b/dataclasses_avroschema/utils.py
@@ -4,6 +4,7 @@ import typing
 from datetime import datetime, timezone
 from functools import lru_cache
 
+import typing_extensions
 from typing_extensions import Annotated, get_origin
 
 from .protocol import ModelProtocol  # pragma: no cover
@@ -20,6 +21,24 @@ try:
     import faust  # pragma: no cover
 except ImportError:  # type: ignore # pragma: no cover
     faust = None  # type: ignore # pragma: no cover
+
+
+@lru_cache(maxsize=None)
+def _get_typing_objects_by_name_of(name: str) -> tuple[typing.Any, ...]:
+    """Get the member named `name` from both `typing` and `typing-extensions` (if it exists)."""
+    result = tuple(getattr(module, name) for module in (typing, typing_extensions) if hasattr(module, name))
+    if not result:
+        raise ValueError(f'Neither `typing` nor `typing_extensions` has an object called {name!r}')
+    return result
+
+
+def _is_typing_name(obj: object, name: str) -> bool:
+    """Return whether `obj` is the member of the typing modules (includes the `typing-extensions` one) named `name`."""
+    # Using `any()` is slower:
+    for thing in _get_typing_objects_by_name_of(name):
+        if obj is thing:
+            return True
+    return False
 
 
 @lru_cache(maxsize=None)
@@ -82,8 +101,16 @@ def is_self_referenced(a_type: typing.Type, parent: typing.Type) -> bool:
 
 
 def is_annotated(a_type: typing.Type) -> bool:
-    origin = get_origin(a_type)
-    return origin is not None and isinstance(origin, type) and issubclass(origin, Annotated)  # type: ignore[arg-type]
+    """
+    Given a python type, return True if is typing.Annotated, otherwise False
+
+    Arguments:
+        a_type (typing.Any): python type
+
+    Returns:
+        bool
+    """
+    return _is_typing_name(get_origin(a_type), name="Annotated")
 
 
 def rebuild_annotation(a_type: typing.Type, field_info: FieldInfo) -> typing.Type:

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -15,6 +15,7 @@ classifiers = [
     "Programming Language :: Python :: 3.10",
     "Programming Language :: Python :: 3.11",
     "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: 3.13",
     "Programming Language :: Python :: 3 :: Only",
     "Topic :: Software Development",
 ]

